ios/Gekidou/Sources/Gekidou/PushNotification/PushNotification+Signature.swift

```diff
@@ -0,0 +1,133 @@
+import Foundation
+import UserNotifications
+import os.log
+import SwiftJWT
+
+struct NotificationClaims : Claims {
+    var ack_id: String;
+    var device_id: String;
+}
+
+extension PushNotification {
+    public func verifySignatureFromNotification(_ notification: UNMutableNotificationContent) -> Bool {
+        return self.verifySignature(notification.userInfo)
+    }
+    public func verifySignature(_ userInfo: [AnyHashable : Any]) -> Bool {
+        guard let signature =  userInfo["signature"] as? String
+        else {
+            os_log(
+                OSLogType.default,
+                "Mattermost Notifications: Signature verification: No signature in the notification"
+            )
+            return true
```

Comment:
I'm assuming this is backwards compatibility related, am i right?

Reply:
Yes. This is the "old pushproxy" scenario for backwards compatibility.